File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.java
class Solution {
    public int[][] construct2DArray(int[] arr, int m, int n) {
        if (arr.length != m*n) return new int[0][0];
        int[][] ans = new int[m][n];

        int a = 0, b = 0, c = 0;
        for (int i : arr){
            c++;
            ans[a][b++] = i;
            if (c % n == 0) {
                a++;
                b = 0;
            }
        }

        return ans;
    }
}

File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        vector<vector<int>> newArr(m,vector<int>(n));
        if(m*n!=original.size())
            return {};
        else{
            int k=0;
            for(int i=0;i<m;i++){
                
                for(int j=0;j<n;j++){
                    newArr[i][j]=original[k++];
                    
                }
                
            }
        }
        return newArr;
    }
};